### PR TITLE
The list of apps  should be sorted when printed

### LINKED ISSUE #282: Sort list of apps for arkade get
## Expected Behaviour

The list of apps should be sorted when printed, regardless of how they are structured in the code

## Current Behaviour

The list is not ordered so it may be harder to find what you want

## Possible Solution

Use sorting in Go whilst printing - https://golang.org/pkg/sort/

### PR DESCRIPTION
**Pull request by developer-guy**
## Description
The list of apps should be sorted when printed

## Motivation and Context

Fixes #282 .
- [x] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))

## How Has This Been Tested?
After the development is done, I build the code with the target "all" of the Makefile and checked the output whether sorted or not.
This is the screenshot of the output after the sort feature adapted
Screen Shot 2020-11-29 at 22 02 51

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

- [ ] I have tested this on arm, or have added code to prevent deployment

### CODE REVIEW

File: pkg/get/tools.go [hunk 1]
```diff
@@ -1,6 +1,25 @@
 package get
 
-func MakeTools() []Tool {
+import "strings"
+
+func (t Tools) Len() int { return len(t) }
```
Comment (alexellis) [inline pkg/get/tools.go:5]: Given that you're sorting the tools themselves, can you also add a unit test please?

Just create some fake data perhaps with three values and show that the order is as expected.

**Review by alexellis — CHANGES_REQUESTED** [review]
Looks good. Please can you add a unit test for the sorting?

File: pkg/get/tools.go (hunk 1 above)
Comment (developer-guy) [reply pkg/get/tools.go:5]: Thanks for your comment, done.

**Comment by developer-guy** [comment]
> Looks good. Please can you add a unit test for the sorting?

Thank you, of course, I added a unit test for the sorting feature.

File: pkg/get/get_test.go [hunk 2]
```diff
@@ -28,6 +30,50 @@ func getTool(name string, tools []Tool) *Tool {
...
+			Name:  "helmfile",
+		},
+		{
+			Owner: "kubernetes",
+			Repo:  "kubernetes",
+			Name:  "kubectl",
+		},
+	}
+
+	if !reflect.DeepEqual(got, want) {
```
Comment (alexellis) [inline pkg/get/get_test.go:72]: For the size of the test, I would prefer explicit comparisons of each index, but will take this.

**Review by alexellis — APPROVED** [review]
Approved

**Comment by developer-guy** [comment]
> Approved

thank you 🙏

**Comment by alexellis** [comment]
This made it into the latest release, congrats. https://github.com/alexellis/arkade/releases/tag/0.6.25

**Pull request merged**